### PR TITLE
Descriptor unit tests

### PR DESCRIPTION
**Pull request by bartoszwilk**

### CODE REVIEW

File: test/test_util/characteristic_generator.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import 'dart:convert';
+
+import 'package:flutter_ble_lib/flutter_ble_lib.dart';
+import 'package:flutter_ble_lib/src/_managers_for_classes.dart';
+
+class CharacteristicGenerator {
```
[the remark this answers is not in the document]
Comment (bartoszwilk) [reply test/test_util/characteristic_generator.dart]: Actually I cherry picked the commit with both generators from the branch with service tests. It's used there.

File: test/ble_manager_test.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+void main() {
+  test('Ble manager must be singleton', () {
+    //given
+    final firstInstanceOfBlemanager = BleManager();
+
+    //when
+    final secondInstanceOfBleManager = BleManager();
+
+    //then
+    expect(secondInstanceOfBleManager, firstInstanceOfBlemanager);
```
Comment (mikolak) [inline test/ble_manager_test.dart]: Different matcher should be used: https://api.flutter.dev/flutter/package-matcher_matcher/same.html

File: test/descriptor_test.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    verify(
+      managerForDescriptor.readDescriptorForIdentifier(descriptor, "456"),
+    );
+  });
+
+  test(
+      "read reads descriptor's value with generated transactionId using manager",
+      () {
+    //when
+    descriptor.read(transactionId: null);
```
[the remark this answers is not in the document]
Comment (bartoszwilk) [reply test/descriptor_test.dart]: Good point

File: test/descriptor_test.dart [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:flutter_ble_lib/flutter_ble_lib.dart';
+import 'package:flutter_ble_lib/src/_managers_for_classes.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'mock/mocks.dart';
```
Comment (mikolak) [inline test/descriptor_test.dart]: Do we need a separate directory for one file?

**Pull request merged**